quix-frontend/client/src/lib/sql-autocomplete/adapter/sql-autocpmplete-adapter-utills.ts

```diff
@@ -0,0 +1,304 @@
+  if(!prefix.includes('.')) {
+    return [];
+  }
+  const allChildren = getAllChildrenOfTables(tables);
+  const relevantPartOfPrefix = findRelevantPartOfPrefix(tables , prefix.split('.')).slice(0, -1);
+  if( !relevantPartOfPrefix )  {
+    return []
+  }
+  const lastDotIndex = relevantPartOfPrefix.lastIndexOf('.');
+  const startOfSearch = lastDotIndex !== -1 ? relevantPartOfPrefix.slice(0, lastDotIndex + 1) : relevantPartOfPrefix;
+  const searchPart = relevantPartOfPrefix.replace(startOfSearch,'')
+  const filteredChildren = allChildren.filter(obj => {
+  const nameInLowerCase = obj.name.toLowerCase();
+  const parts = nameInLowerCase.split('.');
+    if (parts.length > 1) {
+        const substringAfterFirstDot = parts.slice(1).join('.');
+        const criteria = checkCriteria(substringAfterFirstDot , searchPart.toLowerCase());
+        const flterIfInDiffrenCollumn = obj.name.startsWith(prefix.split('.')[0]);
+        return obj.name.includes(startOfSearch) && substringAfterFirstDot.includes(searchPart.toLowerCase()) && criteria && flterIfInDiffrenCollumn ;
+    }
+    return false;
+});
+const prefixUntilLastDot = extractPrefixUntilLastDot(prefix) ;
+  const completionArray = filteredChildren.map(obj => ({
+    value: obj.name,
+    meta : typeof obj.dataType === 'object' ? 'row' : obj.dataType,
+    caption:  obj.name.slice( prefixUntilLastDot.length +1)
+  }));
+  return completionArray;
+}
+
+export function getNextLevel(tables: TableInfo[] , prefix: string | undefined): any {
+  const allChildren = getAllChildrenOfTables(tables);
+  const relevantPartOfPrefix = findRelevantPartOfPrefix(tables , prefix.split('.')).slice(0, -1); //if same problem for both change in function itself
+  const relevantChildren = allChildren.filter(obj => {
+  const dotCount = obj.name.split('.').length - 1;
+    return obj.name.startsWith(relevantPartOfPrefix) && dotCount === relevantPartOfPrefix.split('.').length - 1;
+  });
+  const completionArray = relevantChildren.map(obj => ({
+    value: prefix.replace(relevantPartOfPrefix, '') + obj.name,
+    meta : typeof obj.dataType === 'object' ? 'row' : obj.dataType,
+    caption:  obj.name.slice(relevantPartOfPrefix.length)
+  }));
+  return completionArray
+}
+
+function checkCriteria(substringAfterFirstDot, searchPart) {
+  const lastDotIndex = substringAfterFirstDot.lastIndexOf(".");
+  
+  if (lastDotIndex !== -1) {
+      const partAfterLastDot = substringAfterFirstDot.substring(lastDotIndex + 1);
+      if (partAfterLastDot.includes(searchPart)) {
+          return true;
+      }
+  } else if (substringAfterFirstDot.includes(searchPart)) {
+      return true;
+  }
+  
+  return false;
+}
+
+function extractPrefixUntilLastDot(inputString) {
+  const lastDotIndex = inputString.lastIndexOf(".");
+  
+  if (lastDotIndex !== -1) {
+      const prefix = inputString.substring(0, lastDotIndex);
+      return prefix;
+  }
+  
+  return inputString;
+}
```

Comment:
you created this code? or via GPT?

Reply:
some people call me that

Reply:
This file is complex file.
300 lines is not given.
Consider wrap it in class + separate another file.

Reply:
seperated to 2 diffrent files